pyproject.toml

```diff
@@ -63,45 +65,46 @@ mkdocstrings = { version = ">=0.23.0" }
 black = "^24.4.2"
 pytest-doctestplus = "^1.2.1"
 
-[tool.poetry.urls]
-"Bug Tracker" = "https://github.com/fulcrumgenomics/prymer/issues"
-
 [build-system]
 requires      = ["poetry-core>=1.6"]
 build-backend = "poetry.core.masonry.api"
 
-[tool.git-cliff.changelog]
-header = ""
-trim = true
-body = """
-{% for group, commits in commits | group_by(attribute="group") %}
-    ## {{ group | upper_first }}
-    {% for commit in commits %}
-        - {{ commit.message | upper_first }} ({{ commit.id | truncate(length=8, end="") }})\
-    {% endfor %}
-{% endfor %}\n
-"""
+[tool.poe]
+poetry_command = "task"
 
-[tool.git-cliff.git]
-conventional_commits = true
-commit_parsers = [
-    { message = "^.+!:*", group = "Breaking"},
-    { message = "^feat*", group = "Features"},
-    { message = "^fix*", group = "Bug Fixes"},
-    { message = "^docs*", group = "Documentation"},
-    { message = "^perf*", group = "Performance"},
-    { message = "^refactor*", group = "Refactor"},
-    { message = "^style*", group = "Styling"},
-    { message = "^test*", group = "Testing"},
-    { message = "^chore\\(release\\):*", skip = true},
-    { message = "^chore*", group = "Miscellaneous Tasks"},
-    { body = ".*security", group = "Security"}
+[tool.poe.tasks]
+fix-format = "ruff format"
+fix-lint = "ruff check --fix"
+
+fix-all.ignore_fail = true
+fix-all.sequence = [
+    "fix-format",
+    "fix-lint"
+]
+
+check-lock = "poetry check --lock"
+check-format = "ruff format --check --diff"
+check-lint = "ruff check"
+check-tests = "pytest"
+check-typing = "mypy"
+
+check-all.ignore_fail = true
+check-all.sequence = [
+    "check-lock",
+    "check-format",
+    "check-lint",
+    "check-tests",
+    "check-typing"
+]
+
+fix-and-check-all.ignore_fail = true
+fix-and-check-all.sequence = [
+    "fix-all",
+    "check-all"
 ]
-filter_commits = false
 
 [tool.mypy]
```

Comment:
There are additional config in the Python template that will make MyPy stricter but I did not add it because it found a lot of additional typing issues. I suggest running MyPy in an even stricter mode once this branch stabilizes more.